### PR TITLE
Feature/MusicAlbum

### LINKED ISSUE #30: [ 1pts ] Create MusicAlbum class in a separate .rb file.
Create a new file with the class MusicAlbum

### LINKED ISSUE #31: [ 1pts ] Create Genre class with an association to the Item class (in a separate .rb file).
Create a new file with the class Genre connected to the Item class.

### LINKED ISSUE #32: [ 1pts ] Initialize  MusicAlbum Class
All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #33: [ 1pts ] Initialize Genre class
All Genre class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #35: [ 1pts ] Implement methods add_item
add_item method in the Genre class should add the input item to the collection of items

### LINKED ISSUE #36: [ 1pt] Implement methods add_item
add_item method in the Genre class  should add self as a property of the item object (by using the correct setter from the item object)

### LINKED ISSUE #37: Implement methods: can_be_archived?()
[ 1pts ] can_be_archived?() in the MusicAlbum class should override the method from the parent class

### LINKED ISSUE #38: [ 1pts ] Implement methods: can_be_archived?()
can_be_archived?() in the MusicAlbum class should return true if parent's method returns true AND if on_spotify equals true otherwise, it should return false

### LINKED ISSUE #39: Method can_be_archived?()
can_be_archived?() in the MusicAlbum class should return true if parent's method returns true AND if on_spotify equals true otherwise, it should return false

### LINKED ISSUE #40: [ 1pts ] Add unit tests for all implemented methods.
Test every method with unit tests

### LINKED ISSUE #50: [ 1pts ]  Options avaiable
The following options should be available:

 - List all music albums
 -  List all genres
 -  Add a music album

### LINKED ISSUE #61: [ 1pt] Implement methods
add_item method in the Genre class should take an instance of the Item class as an input

### PR DESCRIPTION
**Pull request by AmaliaGM**
# Issues closed : #30 - #31 - #32 - #33 - #34 - #35 - #36 - #37 - #38 - #39 - #50
- Create MusicAlbum class in a separate .rb file.
- Create Genre class with an association to the Item class (in a separate .rb file).
- All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
add_item method in the Genre class
should take an instance of the Item class as an input
should add the input item to the collection of items
should add self as a property of the item object (by using the correct setter from the item object)
can_be_archived?() in the MusicAlbum class
should override the method from the parent class
should return true if parent's method returns true AND if on_spotify equals true
otherwise, it should return false

- close #30 
- close #31 
- close #32 
- close #33 
- close #35 
- close #36 
- close #37 
- close #38 
- close #39
- close #40
- close #50
- close #61

### CODE REVIEW

File: src/genre.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative './item'
+require 'secure_random'
+
+class Genre < Item
```
Comment (carlosmuerto) [inline src/genre.rb:4]: Genre does not inherit from Item, Genre is a component of Item

File: src/genre.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  initialize(id, name, items, input)
+  @id || SecureRandom.random_number(1000)
+  @name = name
+  @items = []
+  @add_item = input
+end
+
+def add_item(_items)
+  @items << item
+  @items.push(item)
```
Comment (carlosmuerto) [inline src/genre.rb:16]: did you are adding twice to the items list?

File: src/music-album.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+require './item'
+
+class MusicAlbum < Item
+  attr_accesor :on_spotify, :id
+
+  def initialize(on_spotify, *args)
+    super(*args)
```
Comment (carlosmuerto) [inline src/music-album.rb]: the arguments of the Item are not optional add all to the constructor 

```ruby
def initialize(  # rubocop:disable Metrics/ParameterLists
  genre, 
  author, 
  source, 
  label, 
  publish_date, 
  on_spotify
) 
  super(genre, author, source, label, publish_date)
  @on_spotify = on_spotify
end
```

File: src/genre.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  attr_accesor :id, :name, :items, :add_item
+
+  initialize(id, name, items, input)
+  @id || SecureRandom.random_number(1000)
+  @name = name
+  @items = []
+  @add_item = input
+end
+
+def add_item(_items)
```
Comment (carlosmuerto) [inline src/genre.rb:14]: Add_item(item) method need to be called when a item is constructed

**Review by carlosmuerto — CHANGES_REQUESTED** [review]
There is some change, please read the comments:
- please add the issues you are closing in the PR description like this` - close #<number of the issue>`

**Review by carlosmuerto — CHANGES_REQUESTED** [review]
please use the ` - close #<number of the issue>` format to identify what issue are your addressing
it will show like this

- close #30

**Review by andym80 — CHANGES_REQUESTED** [review]
- [ ] Check

**Comment by AmaliaGM** [comment]
> * [ ]  Check

No se ven tus comentarios

**Review by carlosmuerto — APPROVED** [review]

**Review by andym80 — APPROVED** [review]

**Pull request merged**